model_compression_toolkit/core/keras/mixed_precision/configurable_weights_quantizer.py

```diff
@@ -0,0 +1,154 @@
+                        self.node_q_cfg[0].weights_quantization_cfg.enable_weights_quantization:
+                    Logger.error("Candidates with different weights enabled properties is currently not supported.")
+
+            # Initialize quantized weights for each weight that should be quantized.
+            self.quantized_weights = []
+            for qc in self.node_q_cfg:
+                qc_weights = qc.weights_quantization_cfg
+                q_weight = qc_weights.weights_quantization_fn(self.float_weights,
+                                                              qc_weights.weights_n_bits,
+                                                              True,
+                                                              qc_weights.weights_quantization_params,
+                                                              qc_weights.weights_per_channel_threshold,
+                                                              qc_weights.weights_channels_axis)
+
+                self.quantized_weights.append(tf.convert_to_tensor(q_weight, dtype=tf.float32))
+
+            self.active_quantization_config_index = self.max_candidate_idx
+
+        def set_weights_bit_width_index(self,
+                                        index: int):
+            """
+            Change the "active" bitwidth index the configurable quantizer uses, so a different quantized weight
+            will be used.
+
+            Args:
+                index: Quantization configuration candidate index to use.
+
+            """
+
+            assert index < len(self.node_q_cfg), \
+                f'Quantizer has {len(self.node_q_cfg)} ' \
+                f'possible nbits. Can not set index {index}'
+            self.active_quantization_config_index = index
+
+        def __call__(self,
+                     inputs: tf.Tensor,
+                     training: bool) -> tf.Tensor:
+            """
+            Method to return the quantized weight. This method is called when the framework needs to quantize a
+            float weight, and is expected to return the quantized weight. Since we already quantized the weight in
+            all possible bitwidths, we do not quantize it again, and simply return the quantized weight according
+            to the current active_quantization_config_index.
+
+            Args:
+                inputs: Input tensor (not relevant since the weights are already quantized).
+                training: Whether the graph is in training mode (not used in this function).
+
+            Returns:
+                Quantized weight, that was quantized using number of bits that is in a
+                specific quantization configuration candidate (the candidate's index is the
+                index that is in active_quantization_config_index the quantizer holds).
+            """
+
+            return self.quantized_weights[self.active_quantization_config_index]
+
+        def get_config(self) -> Dict[str, Any]:  # pragma: no cover
```

Comment:
No one **tests** get config. I don't think it is important to test this, so if there is no intention to write a dedicated test for this method, it can be marked with "no cover".

Reply:
so who calls it?

Reply:
it's part of the base quantizer interface, I believe it is mainly used for serialization, but anyway it should be implemented in any class that inherits from BaseKerasInferableQuantizer